### PR TITLE
feat(attendee registration shortcode): Change the shortcode and option to use page ID

### PR DESCRIPTION
**Pull request by Camwyn**
...instead of page slug.

Change the option to use page ID to identify the page set as folks can change the page slug later.

:ticket: https://central.tri.be/issues/124997

Requires: https://github.com/moderntribe/event-tickets-plus/pull/760

### CODE REVIEW

File: src/Tribe/Attendee_Registration/Main.php [hunk 1]
```diff
@@ -29,11 +29,18 @@ class Tribe__Tickets__Attendee_Registration__Main {
...
-		$page = get_page_by_path( $slug );
+		$id = Tribe__Settings_Manager::get_option( 'ticket-attendee-page-id', false );
+
+		// check for old saved slug - since TBD
+		if ( ! empty( $id ) ) {
+			$page = get_page( $id );
+			$slug = $page->post_name;
+		} else {
+			$slug = Tribe__Settings_Manager::get_option( 'ticket-attendee-page-slug', false );
+			$page = get_page_by_path( $slug );
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Main.php]: Probably shouldn’t call this function if it doesn’t have a slug

File: src/Tribe/Attendee_Registration/Main.php [hunk 2]
```diff
@@ -29,11 +29,18 @@ class Tribe__Tickets__Attendee_Registration__Main {
...
+		}
 
-		if ( empty( $slug ) || ! has_shortcode( $page->post_content, 'tribe_attendee_registration' ) ) {
+		if ( empty( $slug ) || empty( $page ) || ! has_shortcode( $page->post_content, 'tribe_attendee_registration' ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Main.php]: Will there be a case where the isn’t both or none: slug/page?

Comment (Camwyn) [reply src/Tribe/Attendee_Registration/Main.php]: eh, trying to avoid calling `$page->post_content` if we don't have a `$page` - let me rework that.

**Comment by sc0ttkclark** [comment]
Let's chat a little more about backcompat

**Pull request merged**